src/net.h

```diff
@@ -10,6 +10,7 @@
 #include "compat.h"
 #include "limitedmap.h"
 #include "netbase.h"
+#include "primitives/block.h"
```

Comment:
I have also thought about this during the time of writing the code. I think its more flexible the way it is now. `FilterStatsProcessBlock` could also count transactions or the total amount of vins/vouts per block (which is basically where the filter needs to "go through").

Including `block.h` should be not a problem.